### PR TITLE
Test <dialog> + shadow DOM focus

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/7285

### CODE REVIEW

**Review by sefeng211 — COMMENTED** [review]
The tests look good and reasonable. 

What do you think of adding another test for a nested shadow dom with the `delegates focus` attribute, because `autofocus` can take precedence again in this case?

**Comment by domenic** [comment]
Thanks for the review! Good idea on the extra test; I added it and I think it's correct. (It sadly was a bit too much for my generic declarative testing framework, so I put it in a separate file.)

File: html/semantics/interactive-elements/the-dialog-element/dialog-focus-shadow-delegatesfocus.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  template.replaceWith(div);
+}
+
+for (const template of document.querySelectorAll(".turn-into-shadow-tree")) {
+  turnIntoShadowTree(template);
+}
+
+for (const method of ["show", "showModal"]) {
+  test(t => {
+    const dialog = document.querySelector("dialog");
```
Comment (sefeng211) [inline html/semantics/interactive-elements/the-dialog-element/dialog-focus-shadow-delegatesfocus.html:39]: `dialog[method]();` is missing?

Comment (domenic) [reply html/semantics/interactive-elements/the-dialog-element/dialog-focus-shadow-delegatesfocus.html:39]: Oops, thanks!

**Review by sefeng211 — APPROVED** [review]
LGTM!

**Comment by josepharhar** [comment]
I may have followup questions when I actually implement the behavior in chrome, but LGTM!

**Pull request merged**